operator/controllers/kyma_controller.go

```diff
@@ -212,7 +231,7 @@ func (r *KymaReconciler) HandleProcessingState(ctx context.Context, kyma *v1alph
 
 	// if the ready condition is not applicable, but we changed the conditions, we still need to issue an update
 	if statusUpdateRequiredFromCreation || statusUpdateRequiredFromSync {
```

Comment:
The reason why this is done is to prevent too many reconciliation loops, otherwise we would have to reconcile n times for 1 kyma with n modules for the status updates. Ideally I wanted to avoid that, and the easiest way to solve the issue at the time was to just collect all updates and return at the end. That way I reduce the amount of reconciliations heavily. Ideally I would like to stick with that

Reply:
refactored to do update in the end with additional `statusUpdateRequiredFromDeletion`